cmd/juju/machine/upgradeseries.go

```diff
@@ -207,33 +213,30 @@ func (c *upgradeSeriesCommand) Run(ctx *cmd.Context) error {
 // dependency this function should contain minimal logic other than gathering an
 // API handle and making the API call.
 func (c *upgradeSeriesCommand) UpgradeSeriesPrepare(ctx *cmd.Context) error {
-	var apiRoot api.Connection
-
-	// If the upgradeMachineSeries is nil then we collect a handle to the
-	// API. If it is not nil it is likely the client has been set elsewhere
-	// (i.e. a test mock) so we don't reset it.
-	if c.upgradeMachineSeriesClient == nil {
-		var err error
-		apiRoot, err = c.NewAPIRoot()
-		if err != nil {
-			return errors.Trace(err)
-		}
+	apiRoot, err := c.ensureAPIClients()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if apiRoot != nil {
 		defer apiRoot.Close()
-		c.upgradeMachineSeriesClient = machinemanager.NewClient(apiRoot)
 	}
 
-	err := c.promptConfirmation(ctx)
+	units, err := c.promptConfirmation(ctx)
```

Comment:
Looks good! However, should a command which simply handles prompting return a list of units. I am unsure about this. Perhaps we can get prompt confirmation to accept the units as arguments. Since now the units are used elsewhere perhaps we can pull this out. Is this possible?